app/plugin.go

```diff
@@ -182,6 +187,74 @@ func (a *App) InitPlugins(pluginDir, webappPluginDir string) {
 	a.SyncPluginsActiveState()
 }
 
+// SyncPlugins synchronizes the plugins installed locally
+// with the plugin bundles available in the file store.
+func (a *App) SyncPlugins() *model.AppError {
+	mlog.Info("Synching plugins with the file store")
+
+	pluginDirs, err := ioutil.ReadDir(*a.Config().PluginSettings.Directory)
+	if err != nil {
+		return model.NewAppError("SyncPlugins", "app.plugin.sync.read_local_folder.app_error", nil, err.Error(), http.StatusInternalServerError)
+	}
+
+	// Remove plugins locally to sync with file store.
+	for _, dir := range pluginDirs {
+		if !dir.IsDir() {
+			continue
+		}
+
+		// Only handle managed plugins with .filestore flag file.
+		_, err := os.Stat(filepath.Join(*a.Config().PluginSettings.Directory, dir.Name(), managedPluginFileName))
+
+		if os.IsNotExist(err) {
+			mlog.Warn("Found unmanaged plugin. Ignoring in plugin sync with the filestore.", mlog.String("plugin", dir.Name()))
+
+		} else if err != nil {
+			mlog.Error("Error reading local plugin directoy. Skipped for sync.", mlog.String("folder", dir.Name()), mlog.Err(err))
+
+		} else {
+			mlog.Debug("Plugin Sync: Uninstalling plugin locally", mlog.String("plugin", dir.Name()))
+			if err := a.removePluginLocally(dir.Name()); err != nil {
+				mlog.Error("Plugin Sync: Error uninstalling managed plugin.", mlog.String("plugin", dir.Name()), mlog.Err(err))
```

Comment:
Would not recommend using `:` character in the error string as `https://godoc.org/github.com/pkg/errors` (which is used across the repo) uses `:` to wrap and build error stack.
https://github.com/pkg/errors#retrieving-the-cause-of-an-error

`Error uninstalling managed plugin.` should suffice i think, unless others have thoughts on this 🤷‍♂

Reply:
Fixed